### PR TITLE
(531) Replaced Contract handling with BusinessLinks

### PR DESCRIPTION
**Pull request by kekonen**
#531 
Changed the event subscription from 
- `einvoice-send.inChannelContract.updated` / `einvoice-send.inChannelContract.created` 

to
- business-link.business-link.updated

### CODE REVIEW

File: src/server/workflow/workflow.js [hunk 1]
```diff
@@ -686,24 +693,56 @@ module.exports = function(app, db) {
...
+                    reference[businessLink.customerId] = statusesFromContact;
+                    const statusMap = Object.values(statusesFromContact);
+                    if (statusMap.every(el => el == 'connected')){
+                        const [count] = await contact.update({reference, status: 'connected'});
+                    } else if (statusMap.every(el => el == 'rejected')) {
+                        const [count] = await contact.update({reference, status: 'rejected'});
+                    } else if (statusMap.some(el => el == 'connected')) {
+                        const [count] = await contact.update({reference: reference, status: 'partiallyConnected'});
+                    }
+
```
Comment (nczeranka) [inline src/server/workflow/workflow.js]: "partiallyRejected" as another condition would make sense (not connected, not rejected and not partiallyConnected, but some "rejections")

File: src/server/workflow/workflow.js [hunk 2]
```diff
@@ -686,24 +693,56 @@ module.exports = function(app, db) {
...
+            const contact = await db.models.CampaignContact.findOne({where: { supplierId: businessLink.supplierId }});
+
+            const receivedStatuses = Object.assign({}, ...businessLink.connections.map(connection => ({[connection.type]: connection.status})));
+
+            const reference = JSON.parse(contact.reference);
+
+            const statusesFromContact = reference[businessLink.customerId];
+
+            if (statusesFromContact) {
+                let update = false;
```
Comment (nczeranka) [inline src/server/workflow/workflow.js]: updated or statusChanged

File: src/server/workflow/workflow.js [hunk 3]
```diff
@@ -686,24 +693,56 @@ module.exports = function(app, db) {
...
         try {
-            const [count] = await db.models.CampaignContact.update({
-                status: 'connected'
-            }, {
-                where: {
-                    supplierId: inChannelContract.supplierId,
-                    status: {
-                        $in: allowedTransitions.eInvoiceSupplierOnboarding['connected']
+
+            const contact = await db.models.CampaignContact.findOne({where: { supplierId: businessLink.supplierId }});
```
[the remark this answers is not in the document]
Comment (kekonen) [reply src/server/workflow/workflow.js]: updated

File: src/server/workflow/workflow.js [hunk 4]
```diff
@@ -686,24 +693,56 @@ module.exports = function(app, db) {
...
+
+            const receivedStatuses = Object.assign({}, ...businessLink.connections.map(connection => ({[connection.type]: connection.status})));
+
+            const reference = JSON.parse(contact.reference);
```
[the remark this answers is not in the document]
Comment (kekonen) [reply src/server/workflow/workflow.js]: updated

File: src/server/workflow/workflow.js (hunk 2 above)
Comment (kekonen) [reply src/server/workflow/workflow.js]: updated

File: src/server/workflow/workflow.js (hunk 1 above)
Comment (kekonen) [reply src/server/workflow/workflow.js]: updated

File: src/server/workflow/workflow.js [hunk 5]
```diff
@@ -492,10 +492,17 @@ module.exports = function(app, db) {
                 });
             });
 
+            var reference = {};
```
[the remark this answers is not in the document]
Comment (kekonen) [reply src/server/workflow/workflow.js]: updated

**Pull request merged**